submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,329 @@
+    );
+    if (checkedRadio) checkedRadio.checked = false;
+  }
+
+  selectSort(selected) {
+    const profileNames = document.querySelectorAll('.user-mail');
+    this.searchInput.addEventListener('input', ({ target }) => {
+      let filter = target.value.toUpperCase();
+      profileNames.forEach(nameNode => {
+        if (nameNode.textContent.toLocaleUpperCase().startsWith(filter)) {
+          nameNode.closest('.user-card').hidden = false;
+        } else {
+          nameNode.closest('.user-card').hidden = true;
+        }
+      });
+    });
+
+    const sort = {
+      'sort-gender': () => this.sortByGender(selected.id),
+      'sort-name': () => {
+        this.unckeckedRadioButton('sort-age');
+        this.sortByContent('.first-name', selected.id, this.sortByNameTypes);
+      },
+      'sort-age': () => {
+        this.unckeckedRadioButton('sort-name');
+        this.sortByContent('.age', selected.id, this.sortByAgeTypes);
+      },
+      default: () => false,
+    };
+    return (sort[selected.name] || sort['default'])();
+  }
+
+  startFilter() {
+    this.filtersPanel.addEventListener('click', ({ target }) => {
+      if (target.type === 'radio' || target.type === 'text') {
+        loadMoreData.loadDataByScroll(true);
+        this.selectSort(target);
+      }
+    });
+  }
+
+  sortByGender(key) {
+    const cards = document.querySelectorAll('.user-card');
+    if (key === 'allGender') {
+      Array.from(cards).map(node => node.classList.remove('hidden'));
+    } else {
+      Array.from(cards).find(node => {
+        const currentGender = node.querySelector('.gender');
+        if (currentGender.textContent !== key) {
+          currentGender.closest('.user-card').classList.add('hidden');
+        } else {
+          currentGender.closest('.user-card').classList.remove('hidden');
+        }
+      });
+    }
+  }
+
+  insertSortedProfiles(sortedCards, param) {
+    sortedCards.forEach(card =>
+      this.profilesContainer.insertBefore(card, this.profilesContainer[param])
+    );
+  }
+
+  sortBySelector(selectorName) {
+    const cards = document.querySelectorAll('.user-card');
+    if (selectorName === 'all') {
+      return Array.from(cards).sort((a, b) => a.id.localeCompare(b.id));
+    }
+    return Array.from(cards).sort((a, b) =>
+      a
+        .querySelector(selectorName)
+        .textContent.localeCompare(b.querySelector(selectorName).textContent)
+    );
+  }
+
+  sortByContent(sortSelector, sortType, sortTypes) {
+    const sortedCards = this.sortBySelector(sortSelector);
+    const typeOfSort = {
+      [sortTypes.type1]: () =>
+        this.insertSortedProfiles(sortedCards, 'lastChild'),
+      [sortTypes.type2]: () =>
+        this.insertSortedProfiles(sortedCards, 'firstChild'),
+      [sortTypes.type3]: () =>
+        this.insertSortedProfiles(this.sortBySelector('all'), 'lastChild'),
+      default: () => false,
+    };
+    return (typeOfSort[sortType] || typeOfSort['default'])();
+  }
+}
+
+class Api {
```

Comment:
`API` :)

Reply:
Now it is called class ```Users```

Reply:
API was fine. It's an abbreviation, so it should be written with Caps Lock